concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+) -> None:
+    """
+    Add error if PaymentInfo.amount_paid related to stored PendingResponse of type ForcePaymentCommitted
+    differs from the settlement_payment.amount value.
+    """
+    payment_info = force_payment_committed_pending_response.payments.order_by('id').last()
+    for settlement_payment in matched_objects:
+        if settlement_payment.amount != payment_info.amount_paid:
+            errors.append(
+                Error(
+                    matching_error=MatchingError.MESSAGE_VALUE_DIFFERS_FROM_PAYMENT_VALUE,
+                    message=f'PaymentInfo with pk {payment_info.pk} amount_paid value {payment_info.amount_paid} '
+                            f'related to stored PendingResponse of type ForcePaymentCommitted '
+                            f'with pk {force_payment_committed_pending_response.pk} '
+                            f'differs from the settlement payment with tx_hash {settlement_payment.tx_hash} '
+                            f'amount value {settlement_payment.amount}.',
+                )
+            )
+
+
+def check_error_message_payment_ts_is_different_than_payment_closure_time(
+    errors: List[Error],
+    matched_objects: List,
+    force_payment_committed_pending_response: PendingResponse,
+) -> None:
+    """
+    Add error if PaymentInfo.payment_ts related stored PendingResponse of type ForcePaymentCommitted
+    differs from closure_time of the settlement payment.
+    """
+    payment_info = force_payment_committed_pending_response.payments.order_by('id').last()
+    for settlement_payment in matched_objects:
+        if settlement_payment.closure_time != parse_datetime_to_timestamp(payment_info.payment_ts):
+            errors.append(
+                Error(
+                    matching_error=MatchingError.FORCE_PAYMENT_COMMITED_PAYMENT_TS_DIFFERS_FROM_SETTLEMENT_PAYMENT_CLOSURE_TIMESTAMP,
+                    message=f'PaymentInfo with pk {payment_info.pk} payment ts value {payment_info.payment_ts} '
+                            f'related to stored PendingResponse of type ForcePaymentCommitted '
+                            f'with pk {force_payment_committed_pending_response.pk} '
+                            f'differs from the settlement payment with tx_hash {settlement_payment.tx_hash} '
+                            f'closure time value {settlement_payment.closure_time}.',
+                )
+            )
+
+
+def check_error_more_than_one_subtask_result_of_the_same_type_for_subtask(
+    errors: List[Error],
+    matched_objects: List,
+    additional_verification_payment: CoverAdditionalVerificationEvent,
+) -> None:
+    """ Add error if there is more than two SubtaskResultsSettled or SubtaskResultsRejected for the same subtask. """
+    if (
+        any(
+            [subtask_result.response_type == PendingResponse.ResponseType.SubtaskResultsSettled.name  # pylint: disable=no-member
+             for subtask_result in matched_objects]
```

Comment:
Here you created `[]` a little bit different then in our coding style. You should move comparison and for loop to new line

Reply:
I'd suggest to putting the result of `any()` in a variable with a good name. It will make the `if` more concise and easier to understand what you're checking.

Also, `[]` is not really necessary. `any()` can work with a generator expression too.